app/models/concerns/account_interactions.rb

```diff
@@ -141,6 +156,10 @@ def muting_notifications?(other_account)
     mute_relationships.where(target_account: other_account, hide_notifications: true).exists?
   end
 
+  def muting_reblogs?(other_account)
+    active_relationships.where(target_account: other_account, show_reblogs: false).exists?
```

Comment:
I don't quite understand this. Are you saying to do something like `other_account.show_reblogs`? (If so, that won't work: `show_reblogs` is set on the follow relationship, not on the account object, so different users can have different settings.

In terms of caching, are you suggesting that `where(target_account: other_account).show_reblogs` would be cached already? I don't think it actually will be, but I can investigate if that's what you're suggesting. The existing line has the benefit of being a single query that doesn't require actually loading data from the database, though.

Reply:
Bad phrasing from me, sorry. I meant the relationship.

I was wondering if `active_relationships.where(target_account: other_account)` might be cached already, but if that's not something likely called multiple times in a single request, then this line is fine as-is.

Reply:
Ah, okay. I don't think it's likely to be cached in a way Rails will pull it back out, unfortunately. (In theory, if you're adding a notification for a status, you probably loaded the poster's active relationships. However, even if the relationship is fully annotated, I don't think Rails is smart enough to recognize that in the opposite direction.)

Reply:
It might be if the `inverse_of` is set on the relationship, but I doubt that'd work if you pulled *all* relationships (instead of this particular one). Not sure if worth the effort to investigate either since this query isn't particularly heavy either.